packages/node/src/stack-parser.ts

```diff
@@ -0,0 +1,123 @@
+import { basename, dirname, StackLineParser } from '@sentry/utils';
+
+const mainModule: string = `${
+  (require.main && require.main.filename && dirname(require.main.filename)) || global.process.cwd()
+}/`;
+
+/** Gets the module */
+function getModule(filename: string | undefined, base?: string): string | undefined {
+  if (!filename) {
+    return;
+  }
+
+  if (!base) {
+    // eslint-disable-next-line no-param-reassign
+    base = mainModule;
```

Comment:
Why this if statement over a default parameter?

Reply:
Check out my latest changes. I think the node parser will need to work even when `require` is not available. 

In that case the line parser can be created like: `node(require)`.

Reply:
Is there any environment where that is the case?

Reply:
In the Electron renderer, `require` may or may not be available depending on runtime configuration. 

Obviously, if `require` is not available, we are not going to encounter any node stack frames but then it's something extra for users to configure and more docs are required etc, etc. 

If the parser works well enough without the need for `require` we can include it **by default** in the Electron renderer without the risk of breaking bundlers and users will get node stack frames decoded if they ever appear.

Reply:
I've backed out the funky require thing for now since it appears webpack really doesn't like it

Reply:
Ah I think this is why we have stuff like https://github.com/getsentry/sentry-javascript/blob/6809dd2ae41f81e3b5d3e3c8076f2c6bcb9cfbfb/packages/utils/src/node.ts#L28-L31